packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -3790,6 +3816,11 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin, Res
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
+    final NavigatorState? parentNavigator = context.findAncestorStateOfType<NavigatorState>();
```

Comment:
This won't trigger rebuild when it changes

Reply:
How to proper get the NavigatorState? 

Can be something like this?
```dart 
NavigatorState? get _parentNavigator => context.findAncestorStateOfType<NavigatorState>();
```

or there are some InheritedWidget that expose the NavigatorState?

Reply:
There is currently no way to listen for navigator changes as this pr is the first time that we expose property that requires navigator look up.